### PR TITLE
Nimakms/references

### PR DESCRIPTION
**Pull request by nimakms**
Adding runtime and management API references, as well as supporting .http files.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -10,17 +14,35 @@
...
+# Programming interfaces
 
-## Service definitions
+The service consists of two distinct REST-based Application Programming Interfaces (APIs), respectively for management and runtime operations. Both APIs are based on concepts of tokens, services, vaults and access policy as resources. Currently there are no client libraries available for the said REST-based APIs, but this is something we are thinking about for the future.
 
-[Coming soon]
+## Management APIs on ARM
 
-## Management HTTP APIs - on Azure Resource Manager (ARM)
+The management APIs are primarily meant to enable uniform resource management along with other Azure resources, either through REST-based requests or through deployment templates. They are based on Azure Resource Monitor (ARM) and can be used for programmatic resource mangement actions like read, create, update, delete, list etc. The ARM interface also enables joint template deployment of "Token Vault" and other Azure resoruces for deploying full solutions. The web address for these APIs are often under `https://management.azure.com/subscriptions/[subscription-id]/resourceGroups/[resource-group]/providers/Microsoft.TokenVault/vaults/[token-vault-name]`.
```
Comment (joerob-msft) [inline README.md]: Remove often from this word

File: README.md [hunk 2]
```diff
@@ -10,17 +14,35 @@
...
+See [Runtime API reference](/docs/runtime-api-reference.md) for more details.
 
-## Runtime HTTP APIs
+## Resource concepts
+
+### Token
+
+### Service
+
+### Vault
```
Comment (joerob-msft) [inline README.md]: We don't have vault or access policy on runtime.

File: docs/management-api-reference.http [hunk 3]
```diff
@@ -0,0 +1,153 @@
+#########################################################
+#### Variables
+#########################################################
+
+@subscriptionId = [your-subscription-id]
+@resourceGroup = [your-resource-group-name]
+@armBaseUrl = https://management.azure.com/subscriptions
+@vaultsPath = {{armBaseUrl}}/{{subscriptionId}}/resourceGroups/{{resourceGroup}}/providers/Microsoft.TokenVault/vaults
+@version = ?api-version=2018-08-01-preview
+@vaultName = testVault
```
Comment (joerob-msft) [inline docs/management-api-reference.http]: Make this a varable also with [] or else people will try to create with the same name likely

File: README.md (hunk 1 above)
Comment (nimakms) [reply README.md]: Fixed.

File: README.md (hunk 2 above)
Comment (nimakms) [reply README.md]: It's under Resource concepts, common between managment and runtime

File: docs/management-api-reference.http (hunk 3 above)
Comment (nimakms) [reply docs/management-api-reference.http]: Fixed

**Pull request merged**